app/components/Login/index.jsx

```diff
@@ -20,38 +32,60 @@ export default class Login extends React.Component {
     };
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (!Immutable.is(nextProps.local, this.props.local)) {
+      const user = nextProps.local.get('user');
+      if (user && user.status) {
+        if (user.status.text === this.state.info.text && user.status.type === this.state.info.type) {
+          this.setState({shakeInfo: true});
+          setTimeout(this.setState.bind(this, {shakeInfo: false}), 500);
+        } else {
+          if (user.status.type === 'success') {
+            nextProps.history.pushState(null, '/');
+          } else {
+            this.setState({
+              info: {
+                type: user.status.type,
+                text: user.status.text,
+              },
+            });
+          }
+        }
+      }
+    }
+  }
+
+
+  shouldComponentUpdate() {
+    return true;
+  }
 
   auth = (e) => {
     e.preventDefault();
-    if (this.state.info.type === 'error' && this.state.info.code === 1001) {
-      this.setState({shakeInfo: true});
-      setTimeout(this.setState.bind(this, {shakeInfo: false}), 500);
-      return;
-    }
-    this.setState({
-      info: {
-        code: 1001,
-        type: 'error',
-        text: 'Wrong pass',
-      },
-    });
+    const authData = {
+      login: e.target[0].value,
```

Comment:
I think we have a way to get value with input name attribute
I would be better. What are you think?